src/D2L.CodeStyle.Analyzers/Immutability/ImmutableHelpers.cs

```diff
@@ -190,7 +190,8 @@ internal static class ImmutableHelpers {
 				return;
 			}
 
-			throw new Exception( $"Unknown expression syntax type '{expr.GetType()}' when parsing flags: '{expr}'" );
+			// Ignore this because the user may be typing something.

```

Comment:
This seems good. Everything else (that's finished typing) should be a type error (e.g. `[Immutable( Except = true )]`), so no need to also throw here. 👍